### PR TITLE
SELinux: merge .cil policies and add a lot of comments

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
Without this, we first define the unused `container.process` type in one file, and extend that type in a second file.
With this, there's only one .cil file with only one block to define the type we need, `virt_launcher.process`.
Also used this as an opportunity to cleanup the policy and remove the rule for `cpu_online_t` that we already get from being a domain.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by vladikr** [comment]
One downside to getting rid of the base_container.cil is that any time we could simply replace it with a container base file from https://github.com/containers/udica/tree/master/udica/templates
What is the benefit of merging these two? :)

File: cmd/virt-handler/virt_launcher.cil [hunk 1]
```diff
@@ -1,10 +1,55 @@
+; This is the custom SELinux policy for virt-launcher. This file is hopefully temporary.
+; Applications running in regular container usually have container_t as an SELinux type.
+; However, some applications running in virt-launcher (namely libvirtd) need more permissions.
 (block virt_launcher
-    (blockinherit container)
```
Comment (stu-gott) [inline cmd/virt-handler/virt_launcher.cil:2]: It's possible to have multiple blocks in one .cil file. Is there an advantage to merging these blocks? Is there any potential we'll need entries for other components at some point?

**Comment by jean-edouard** [comment]
> One downside of getting rid of the base_container.cil is that at any time we could simply replace it with a container base file from https://github.com/containers/udica/tree/master/udica/templates, but now we will have to update the fields manually...
> What is the benefit of merging these two? :)

There are multiple reasons, here's a few:
- I don't think we ever want to rebase to the Udica base_container.cil. They added a bunch of objects and permissions we most likely don't need
- Installing cil modules take time, like a dozen seconds. Installing 2 takes twice the time, which is non-negligible in high turnaround environments like development setups.
- base_container.cil installs an unused type, container.process (+ other stuff in recent udica versions), when SELinux policies should be kept as minimal as possible...

File: cmd/virt-handler/virt_launcher.cil (hunk 1 above)
Comment (jean-edouard) [reply cmd/virt-handler/virt_launcher.cil:2]: Yes, it is possible. That would mitigate the second bullet point in my reply to @vladikr (see below). But points 1 and 3 would still be valid.

**Comment by vladikr** [comment]
> > One downside of getting rid of the base_container.cil is that at any time we could simply replace it with a container base file from https://github.com/containers/udica/tree/master/udica/templates, but now we will have to update the fields manually...
> > What is the benefit of merging these two? :)
> 
> There are multiple reasons, here's a few:
> 
> * I don't think we ever want to rebase to the Udica base_container.cil. They added a bunch of objects and permissions we most likely don't need
> * Installing cil modules take time, like a dozen seconds. Installing 2 takes twice the time, which is non-negligible in high turnaround environments like development setups.
> * base_container.cil installs an unused type, container.process (+ other stuff in recent udica versions), when SELinux policies should be kept as minimal as possible...

I agree. Offline you also mentioned other reasons. So I'm fine with us maintaining this for now. 
However, I do want to get a state where we don't need to worry about the container definition itself, but we just need to keep *only* our own policy that expends the container_t.  
I'm not sure how to get there though.

Perhaps, as a first step, we should open an issue for udica to make the relevant changes too? It is still being recommended *everywhere* for anyone who wants to extend their selinux policies...

**Comment by jean-edouard** [comment]
> However, I do want to get a state where we don't need to worry about the container definition itself, but we just need to keep _only_ our own policy that expends the container_t.
> I'm not sure how to get there though.

I'm not sure if it's possible in CIL to say "make a copy of this type and add these perms to it".

> 
> Perhaps, as a first step, we should open an issue for udica to make the relevant changes too? It is still being recommended _everywhere_ for anyone who wants to extend their selinux policies...

I don't think anything we've discussed so far pertains to udica, I just think base_container.cil makes sense for their use-case, but not so much for ours...

**Comment by vladikr** [comment]
If there are no better options I agree that this is the best we can do right now.
/lgtm

**Pull request merged**